safe/gui/tools/people_in_buildings_dialog.py

```diff
@@ -0,0 +1,380 @@
+
+        :param iface: A Quantum GIS QGisAppInterface instance.
+        :type iface: QGisAppInterface
+
+        :param parent: Parent widget of this dialog
+        :type parent: QWidget
+
+        :param dock: Optional dock widget instance that we can notify of
+            changes to the keywords.
+        :type dock: Dock
+        """
+
+        QtGui.QDialog.__init__(self, parent)
+        self.setupUi(self)
+        # Save reference to the QGIS interface and parent
+        self.iface = iface
+        self.parent = parent
+
+        self.load_layers_into_combo_box()
+
+        self.buildingLayerComboBox.currentIndexChanged['QString'].connect(
+            self.handle_building_layer)
+        self.censusLayerComboBox.currentIndexChanged['QString'].connect(
+            self.handle_census_layer)
+
+        self.buttonBox.button(QtGui.QDialogButtonBox.Apply).clicked.connect(
+            self.estimate_people_in_buildings)
+
+        self.progressBar.setMinimum(0)
+        self.progressBar.setMaximum(100)
+
+    def load_layers_into_combo_box(self):
+        """Load the layer options and column options into the combobox.
+        """
+        layer_names = self._get_layer_names()
+        self._update_combobox(self.buildingLayerComboBox, layer_names)
+        self._update_combobox(self.censusLayerComboBox, layer_names)
+        self.handle_building_layer(self.buildingLayerComboBox.currentText())
+        self.handle_census_layer(self.censusLayerComboBox.currentText())
+
+    def _select_layer_by_name(self, layer_name):
+        """Get a layer by name
+        :param layer_name: The layer's name
+        :type layer_name: basestring
+
+        :returns: The layer that has the appropriate name or none
+        :rtype: QgsMapLayer, None
+        """
+        layers = self._get_layers()
+        layer_names = [layer.name() for layer in layers]
+        if not layer_name in layer_names:
+            return None
+        index = layer_names.index(layer_name)
+        return layers[index]
+
+    def _get_layers(self):
+        """Get all layers.
+
+        :returns: All layers currently loaded.
+        :rtype: list
+        """
+        layers = self.iface.legendInterface().layers()
+        return layers
+
+    def _get_layer_names(self):
+        """Get the names of all layers.
+
+        :returns: A list of all layer names
+        :rtype: list
+        """
+        layers = self._get_layers()
+        layer_names = [l.name() for l in layers]
+        return layer_names
+
+    @staticmethod
+    def _get_field_names(layer):
+        """Get all attribute names of the layer.
+
+        :param layer: The layer to be investigated.
+        :type layer: VectorLayer
+
+        :returns: Returns a list of all attribute names of the layer.
+        :rtype: list
+        """
+        field_names = [field.name() for field in layer.pendingFields()]
+        return field_names
+
+    def handle_building_layer(self, layer_name):
```

Comment:
Is this a slot? Normal naming convention would be ``building_layer_changed(self, layer_name)``